src/lib/Microsoft.Health.Fhir.Ingest/Data/MeasurementToEventMessageAsyncCollector.cs

```diff
@@ -39,6 +44,29 @@ public async Task AddAsync(IMeasurement item, CancellationToken cancellationToke
             await _eventHubService.SendAsync(eventList, partitionKey, cancellationToken).ConfigureAwait(false);
         }
 
+        public async Task AddAsync(IEnumerable<IMeasurement> items, CancellationToken cancellationToken = default)
+        {
+            EnsureArg.IsNotNull(items, nameof(items));
+            var submissionTasks = items
+                .GroupBy(m => m.DeviceId == null ? 0 : m.DeviceId.ToLower().GetHashCode() % _partitionCount)
```

Comment:
We shouldn't use the internal GetHashCode, it isn't guaranteed to generate the same hash code across app domains/processes.  The issue is if a different process (either due to scale out or app restart) encounters a device id, the hashcode could be different and result for data for the same device landing in different partitions defeating the purpose of partitioning by device id.

I think this is fine for a quick prototype and validation of the idea but we would need a different hashing solution before committing.  

For more information please see https://docs.microsoft.com/en-us/dotnet/api/system.object.gethashcode?view=net-5.0#remarks specifically the warning section.